### PR TITLE
feat: Added `risesocket` mode

### PR DESCRIPTION
**Pull request by IgnatovFedor**
Added risesocket mode both TCP and UNIX with tests

### CODE REVIEW

File: deeppavlov/utils/socket/socket.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            socket_params.update(model_defaults)
+    return socket_params
+
+
+class SocketServer:
+    """
+    Class with socket server. The data received by the socket is processed in the deeppavlov model. Sends model
+    response back as dictionary with two keys - 'status' and 'payload':
+    response['status']: str - 'OK' if data processed successfully, else - error message.
+    response['payload']: str - model response dumped to bytes. Empty if an error occured.
```
Comment (yoptar) [inline deeppavlov/utils/socket/socket.py]: Add documentation for init parameters and class attributes please

File: deeppavlov/utils/socket/socket.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            host = self._params['host']
+            port = port or self._params['port']
+            self._address_family = socket.AF_INET
+            self._launch_msg = f'{self._params["binding_message"]} http://{host}:{port}'
+            self._bind_address = (host, port)
+        elif self._socket_type == 'UNIX':
+            self._address_family = socket.AF_UNIX
+            self._bind_address = socket_file or self._params['unix_socket_file']
+            if os.path.exists(self._bind_address):
+                os.remove(self._bind_address)
```
Comment (yoptar) [inline deeppavlov/utils/socket/socket.py]: Can `self._bind_address` be a `Path` here? No need to use `os.path` and `os.remove` this way

File: deeppavlov/utils/socket/socket.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    _bind_address: Union[Tuple, str]
+    _launch_msg: str
+    _loop: asyncio.AbstractEventLoop
+    _model: Chainer
+    _params: Dict
+    _socket: socket.socket
+    _socket_type: str
+
+    def __init__(self, model_config: Path, socket_type: str, port: Optional[int] = None,
+                 socket_file: Optional[str] = None):
```
Comment (yoptar) [inline deeppavlov/utils/socket/socket.py]:
```suggestion
                 socket_file: Optional[str] = None) -> None:
```

File: deeppavlov/utils/socket/socket.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+
+class SocketServer:
+    """
+    Class with socket server. The data received by the socket is processed in the deeppavlov model. Sends model
+    response back as dictionary with two keys - 'status' and 'payload':
+    response['status']: str - 'OK' if data processed successfully, else - error message.
+    response['payload']: str - model response dumped to bytes. Empty if an error occured.
+    """
+    _bind_address: Union[Tuple, str]
```
Comment (yoptar) [inline deeppavlov/utils/socket/socket.py]:
```suggestion
    _bind_address: Union[Tuple[str, int], str]
```

File: deeppavlov/utils/socket/socket.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        :param status: response status. 'OK' if no error occurred
+        :param payload: Deeppavlov model result
+        :return bytes: {'status': status, 'payload': payload} dumped as bytes array
+        """
+        resp_dict = jsonify_data({'status': status, 'payload': payload})
+        resp_str = json.dumps(resp_dict)
+        return resp_str.encode('utf-8')
+
+
+def start_socket_server(model_config: Path, socket_type: str, port: Optional[int], file: Optional[str]) -> None:
```
Comment (yoptar) [inline deeppavlov/utils/socket/socket.py]: One should not name a variable `file`

File: deeppavlov/utils/socket/socket.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    async def _wrap_error(self, conn: socket.socket, error: str) -> None:
+        self._log.error(error)
+        await self._loop.sock_sendall(conn, await self._response(error, ''))
+
+    @staticmethod
+    async def _response(status: str, payload) -> bytes:
+        """
+        :param status: response status. 'OK' if no error occurred
+        :param payload: Deeppavlov model result
+        :return bytes: {'status': status, 'payload': payload} dumped as bytes array
```
Comment (yoptar) [inline deeppavlov/utils/socket/socket.py]: We use Google style for docstings

File: deeppavlov/utils/socket/socket.py (hunk 3 above)
Comment (IgnatovFedor) [reply deeppavlov/utils/socket/socket.py]: Done

File: deeppavlov/utils/socket/socket.py (hunk 4 above)
Comment (IgnatovFedor) [reply deeppavlov/utils/socket/socket.py]: Done

File: tests/test_quick_start.py [hunk 7]
```diff
@@ -414,6 +418,59 @@ def interact_api(config_path):
...
+                        buf = s.recv(1024)
+                        s.setblocking(False)
+                        if buf:
+                            data += buf
+                        else:
+                            break
+                except BlockingIOError:
+                    pass
+            resp = json.loads(data)
+            assert resp['status'] == 'OK', f"socket request returned status: {resp['status']} with {config_path}"
```
Comment (yoptar) [inline tests/test_quick_start.py]: Please log `socket_type` also

File: tests/test_quick_start.py [hunk 8]
```diff
@@ -427,11 +484,18 @@ def test_interacting_pretrained_model(self, model, conf_file, model_dir, mode):
     def test_interacting_pretrained_model_api(self, model, conf_file, model_dir, mode):
         if 'IP' in mode:
             self.interact_api(test_configs_path / conf_file)
+        else:
+            pytest.skip("Unsupported mode: {}".format(mode))
+
+    def test_interacting_pretrained_model_socket(self, model, conf_file, model_dir, mode):
+        if 'IP' in mode:
+            for socket_type in ['TCP', 'UNIX']:
```
Comment (yoptar) [inline tests/test_quick_start.py]: We should think on when and if we want to test for UNIX-sockets, as these tests could fail on Windows

File: docs/integrations/socket_api.rst [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+    # squad-client.py
+
+    import json
+    import socket
+
+    socket_payload = {
+        "context": [
+            "All work and no play makes Jack a dull boy",
+            "I used to be an adventurer like you, then I took an arrow in the knee."
```
Comment (yoptar) [inline docs/integrations/socket_api.rst]:
```suggestion
            "I used to be an adventurer like you, then I took an arrow to the knee."
```

**Pull request merged**